core/src/test/scala/org/broadinstitute/dsde/rawls/billing/GoogleBillingProjectLifecycleSpec.scala

```diff
@@ -132,4 +165,126 @@ class GoogleBillingProjectLifecycleSpec extends AnyFlatSpec {
         ArgumentMatchers.eq(SamBillingProjectPolicyNames.owner)
       )
   }
+
+  it should "store the billing profile ID during billing project creation" in {
+    val repo = mock[BillingRepository]
+    val bpm = mock[BillingProfileManagerDAO]
+    val samDAO = mock[SamDAO]
+    val wsmResourceRecordDao = mock[WorkspaceManagerResourceMonitorRecordDao]
+    val bp = new GoogleBillingProjectLifecycle(repo, bpm, samDAO, mock[GoogleServicesDAO])
+
+    val createRequest = CreateRawlsV2BillingProjectFullRequest(
+      RawlsBillingProjectName("fake_project_name"),
+      Some(RawlsBillingAccountName("fake_billing_account_name")),
+      None,
+      None,
+      None,
+      None
+    )
+
+    when(
+      bpm.createBillingProfile(
+        ArgumentMatchers.eq(createRequest.projectName.value),
+        ArgumentMatchers.eq(createRequest.billingInfo),
+        any(),
+        ArgumentMatchers.eq(testContext)
+      )
+    )
+      .thenReturn(profileModel)
+
+    when(
+      samDAO.syncPolicyToGoogle(
+        ArgumentMatchers.eq(SamResourceTypeNames.billingProject),
+        ArgumentMatchers.eq(createRequest.projectName.value),
+        ArgumentMatchers.eq(SamBillingProjectPolicyNames.owner)
+      )
+    ).thenReturn(Future.successful(Map(WorkbenchEmail(userInfo.userEmail.value) -> Seq())))
+
+    when(repo.setBillingProfileId(createRequest.projectName, profileModel.getId))
+      .thenReturn(Future.successful(1))
+
+    doReturn(Future.successful())
+      .when(wsmResourceRecordDao)
+      .create(any)
+
+    Await.result(bp.postCreationSteps(
+                   createRequest,
+                   mock[MultiCloudWorkspaceConfig],
+                   testContext
+                 ),
+                 Duration.Inf
```

Comment:
Do we want to allow an infinite wait here?

Reply:
We seem to do this all over the tests. I honestly don't know what the alternative would be.

Reply:
(Not a blocker to merge) In places where we have to wait in a test, it's usually preferable to set some kind of timeout, considering our expectations on how long something will take plus a sizable buffer to avoid needlessly flaky tests.  That way a test will fail fast(er) if it's stuck, and go onto a retry if those are configured.

Reply:
@marctalbott do you happen to know the best practice for Scala unittests? I think `Duration.Inf` is what we commonly use.

Reply:
In this case, everything is mocked out, so I have no problem with using `Duration.Inf` here. The `Await` is a means to get the value out of the `Future` -- the timeout won't have any impact as we're not really waiting for anything in this test since Mockito will return all `Future`s quickly. 

As for other uses of `Duration.Inf` in Rawls unit tests, I'm sure there are places we're misusing it. Still, I'm not _that_ worried about those places since they're all unit tests and in theory won't be waiting for long. I'd be very concerned if we had infinite waits in our integration tests.